src/ocean/util/container/more/Stack.d

```diff
@@ -1,317 +1,316 @@
-            All rights reserved.
+    Copyright:
+        Copyright (c) 2008 Kris Bell.
+        Some parts copyright (c) 2009-2017 sociomantic Labs GmbH.
+        All rights reserved.
 
-        License:
-            Tango Dual License: 3-Clause BSD License / Academic Free License v3.0.
-            See LICENSE_TANGO.txt for details.
-
-        Version: Initial release: April 2008
-
-        Authors: Kris
+    License:
+        Tango Dual License: 3-Clause BSD License / Academic Free License v3.0.
+        See LICENSE_TANGO.txt for details.
 
 *******************************************************************************/
 
 module ocean.util.container.more.Stack;
 
 import ocean.core.Enforce;
+
 version (UnitTest)
+{
     import ocean.core.Test;
+}
 
-/******************************************************************************
+/*******************************************************************************
 
-        A stack of the given value-type V, with maximum depth Size. Note
-        that this does no memory allocation of its own when Size != 0, and
-        does heap allocation when Size == 0. Thus you can have a fixed-size
-        low-overhead instance, or a heap oriented instance.
+    A stack of the given value-type V, with maximum depth Size. Note
+    that this does no memory allocation of its own when Size != 0, and
+    does heap allocation when Size == 0. Thus you can have a fixed-size
+    low-overhead instance, or a heap oriented instance.
 
-******************************************************************************/
+*******************************************************************************/
 
-struct Stack (V, int Size = 0)
+public struct Stack ( V, int Size = 0 )
 {
-        import ocean.core.Verify;
+    public alias nth         opIndex;
+    public alias slice       opSlice;
+    public alias rotateRight opShrAssign;
+    public alias rotateLeft  opShlAssign;
+    public alias push        opCatAssign;
 
-        alias nth              opIndex;
-        alias slice            opSlice;
-        alias rotateRight      opShrAssign;
-        alias rotateLeft       opShlAssign;
-        alias push             opCatAssign;
+    static if (Size == 0)
+    {
+        private uint depth;
+        private V[]  stack;
+    }
+    else
+    {
+        private uint     depth;
+        private V[Size]  stack;
+    }
 
+    /***************************************************************************
 
-        static if (Size == 0)
-                  {
-                  private uint depth;
-                  private V[]  stack;
-                  }
-               else
-                  {
-                  private uint     depth;
-                  private V[Size]  stack;
-                  }
+        Clear the stack
 
-        /***********************************************************************
+        Returns: pointer to itself for chaining calls
```

Comment:
Is `Returns: <whatever>` without a newline valid ddoc? (It's at least unusual for our style.)

Reply:
it's valid ddoc, but still unusual for our style.

Reply:
It is not that uncommon:

```
$ grep -R -n "Returns: " ./src/ | wc -l
269
```

I have been using it in ocean for quite some time with both `Returns:` and `Throws:` to save some vertical space.

Reply:
Go on then. (Our coding style doc doesn't mention this either way, though by convention it's always included a newline, until you came along with your newfangled ideas.)